libraries/app/application.cpp

```diff
@@ -122,44 +121,19 @@ void application_impl::reset_p2p_node(const fc::path& data_dir)
    _p2p_network->load_configuration(data_dir / "p2p");
    _p2p_network->set_node_delegate(this);
 
+   // passed as one URL per parameter
    if( _options->count("seed-node") )
    {
       auto seeds = _options->at("seed-node").as<vector<string>>();
-      for( const string& endpoint_string : seeds )
-      {
-         try {
-            std::vector<fc::ip::endpoint> endpoints = resolve_string_to_ip_endpoints(endpoint_string);
-            for (const fc::ip::endpoint& endpoint : endpoints)
-            {
-               ilog("Adding seed node ${endpoint}", ("endpoint", endpoint));
-               _p2p_network->add_node(endpoint);
-               _p2p_network->connect_to_endpoint(endpoint);
-            }
-         } catch( const fc::exception& e ) {
-            wlog( "caught exception ${e} while adding seed node ${endpoint}",
-                     ("e", e.to_detail_string())("endpoint", endpoint_string) );
-         }
-      }
+      _p2p_network->add_seed_nodes(seeds, true);
```

Comment:
I didn't get why we connect immediately sometimes but not always. IMHO we should always let p2p layer decide whether and when to connect to a peer, E.G. to honor `max_connections`.

Reply:
As the for loop was basically repeated in three places, I moved it into add_seed_nodes. But this for loop is the only one that calls "connect_to_endpoint" and I could not figure out why. I will attempt more research.

Reply:
Adding this line here is causing threading issues. I am investigating.

Reply:
What if change it to `false`?

Reply:
Changing it to false would change the behavior of the original code, which I believe wouldn't matter too much. I cannot see a benefit to connecting immediately either. I would be happy to delay connecting in all three cases and eliminate the extra parameter. But I feel uncomfortable until I know why it was there.

As for the threading issue, it has been resolved.

Reply:
The code to connect immediately was already there in the first commit, see
https://github.com/bitshares/bitshares-core/blame/8944facd4129e60155debdf1064b0a96bd43965a/libraries/app/application.cpp#L109, perhaps @nathanhourt remembers why? I think it's unnecessary because `connect_to_p2p_network()` is called only a few lines below.
Actually it's the new code added later that has missed it, see https://github.com/bitshares/bitshares-core/blame/dfb9043df54a9f51983ed04014cd9ddc2a216323/libraries/app/application.cpp#L148, which does work, so I think we can keep the logic consistent, aka delay all.